sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -97,7 +98,16 @@ object FileSourceStrategy extends Strategy with Logging {
         dataColumns
           .filter(requiredAttributes.contains)
           .filterNot(partitionColumns.contains)
-      val outputSchema = readDataColumns.toStructType
+      val outputSchema = if (fsRelation.sqlContext.conf.isParquetNestColumnPruning) {
```

Comment:
Oh, I meant enabling/disabling affects the other data sources. I see it's disabled by default. I ran ,for example, `JsonSuite` after enabling this option (after leaving the comment above) and saw some failures related with nested structures.